src/time-picker/time-picker.jsx

```diff
@@ -44,23 +46,35 @@ let TimePicker = React.createClass({
 
   formatTime(date) {
     let hours = date.getHours();
-    let mins = date.getMinutes();
-    let aditional = "";
+    let mins = date.getMinutes().toString();
 
     if (this.props.format === "ampm"){
       let isAM = hours < 12;
       hours = hours % 12;
-      aditional += isAM ? " am" : " pm";
-      hours = hours || 12;
+      let additional = isAM ? " am" : " pm";
+      hours = (hours || 12).toString();
+
+      if (mins.length < 2 ) mins = "0" + mins;
+
+      if (this.props.pedantic) {
+        // Treat midday/midnight specially http://www.nist.gov/pml/div688/times.cfm
+        if (hours === "12" && mins === "00" && additional === " am") {
+          return "12 noon";
+        }
+        if (hours === "12" && mins === "00" && additional === " pm") {
+          return "12 midnight";
+        }
+      }
```

Comment:
This logic is reversed - 12am is midnight and 12pm is noon correct?

Also can be simplified:
```
if (this.props.pedantic && hours === '12' && mins === '00') {
  return additional === ' am' ? '12 midnight' : '12 noon';
}
```